### ISSUE TITLE
E0424 False Positives

### ISSUE THREAD

**Issue by Wolvereness** [intellij-rust/intellij-rust#1795]
These are two false positive cases for the E0424 inspection. The cases are arbitrary and not rustic; they're just for minimal reproduction. `rustc` does not share this false-positive. In particular, the usage of `{}` triggers the false positive, while omitting them and having single imports per line will not trigger the false positive.

```rust
mod e0424 {
    enum T1 { E1, E2 }
    enum T2 { E1, E2 }
    impl T2 {
        pub fn from_t1(t1: &T1) -> Self {
            use self::{T1 as From, T2 as To};
            match t1 {
                &From::E1 => To::E1,
                &From::E2 => To::E2,
            }
        }
    }

    impl T1 {
        pub fn from_t2(t2: &T2) -> Self {
            use self::{T2 as From};
            use self::{T1 as To};
            match t2 {
                &From::E1 => To::E1,
                &From::E2 => To::E2,
            }
        }
    }
}
```

Related: #990 #886.

**Comment by vlad20012** [comment]
@matklad there is one more path resolution problem.

**Comment by vlad20012** [comment]
The relevant code located [here](https://github.com/intellij-rust/intellij-rust/blob/d4a54c17c1dfffe7c7d2cbb8424b0eb126306a20/src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt#L177)

We should check that `path.parent` is not `RsPath` and also not `RsUseItem`
